File: music-api/internal/models/podcast.go
package models

import "time"

// PodcastStatus represents the lifecycle state of a podcast.
type PodcastStatus string

const (
	PodcastStatusDraft     PodcastStatus = "DRAFT"
	PodcastStatusPublished PodcastStatus = "PUBLISHED"
	PodcastStatusArchived  PodcastStatus = "ARCHIVED"
)

// PodcastEpisodeStatus represents the lifecycle state of a
// podcast episode.
//
// SCHEDULED, LIVE, and ENDED are included now so the model
// remains compatible with future scheduled/live podcast work.
type PodcastEpisodeStatus string

const (
	PodcastEpisodeStatusDraft     PodcastEpisodeStatus = "DRAFT"
	PodcastEpisodeStatusScheduled PodcastEpisodeStatus = "SCHEDULED"
	PodcastEpisodeStatusPublished PodcastEpisodeStatus = "PUBLISHED"
	PodcastEpisodeStatusLive      PodcastEpisodeStatus = "LIVE"
	PodcastEpisodeStatusEnded     PodcastEpisodeStatus = "ENDED"
	PodcastEpisodeStatusArchived  PodcastEpisodeStatus = "ARCHIVED"
)

// PodcastEpisodeType describes the editorial role of an episode.
type PodcastEpisodeType string

const (
	PodcastEpisodeTypeFull    PodcastEpisodeType = "FULL"
	PodcastEpisodeTypeTrailer PodcastEpisodeType = "TRAILER"
	PodcastEpisodeTypeBonus   PodcastEpisodeType = "BONUS"
)

// Podcast represents one podcast show/series.
//
// Ownership belongs directly to a user account rather than an
// artist profile.
type Podcast struct {
	ID int64 `json:"id"`

	OwnerUserID int `json:"owner_user_id"`

	Title       string `json:"title"`
	Slug        string `json:"slug"`
	Description string `json:"description"`
	Category    string `json:"category"`

	ArtworkURL      string `json:"artwork_url"`
	ArtworkPublicID string `json:"-"`

	Status PodcastStatus `json:"status"`

	IsExplicit bool `json:"is_explicit"`

	PublishedAt *time.Time `json:"published_at"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// PodcastEpisode represents one episode belonging to a podcast.
type PodcastEpisode struct {
	ID int64 `json:"id"`

	PodcastID int64 `json:"podcast_id"`

	Title       string `json:"title"`
	Slug        string `json:"slug"`
	Description string `json:"description"`

	SeasonNumber  int  `json:"season_number"`
	EpisodeNumber *int `json:"episode_number"`

	EpisodeType PodcastEpisodeType `json:"episode_type"`

	AudioURL      string `json:"audio_url"`
	AudioPublicID string `json:"-"`

	ArtworkURL      string `json:"artwork_url"`
	ArtworkPublicID string `json:"-"`

	DurationMS int64 `json:"duration_ms"`

	Status PodcastEpisodeStatus `json:"status"`

	IsExplicit bool `json:"is_explicit"`

	ScheduledAt *time.Time `json:"scheduled_at"`
	PublishedAt *time.Time `json:"published_at"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// PodcastDetails is the public representation of a podcast
// together with its episodes.
//
// Public API handlers will decide whether the episode collection
// contains only published episodes or owner-visible drafts.
type PodcastDetails struct {
	Podcast Podcast `json:"podcast"`

	Episodes []PodcastEpisode `json:"episodes"`
}

File: music-api/frontend/src/types/podcast.ts
export type PodcastStatus =
  | 'DRAFT'
  | 'PUBLISHED'
  | 'ARCHIVED'

// Live statuses (SCHEDULED, LIVE, ENDED) are set only through
// the live session endpoints — never through episode PATCH.
export type PodcastEpisodeStatus =
  | 'DRAFT'
  | 'PUBLISHED'
  | 'ARCHIVED'
  | 'SCHEDULED'
  | 'LIVE'
  | 'ENDED'

export type PodcastEpisodeType =
  | 'FULL'
  | 'TRAILER'
  | 'BONUS'

export type Podcast = {
  id: number
  owner_user_id: number
  title: string
  slug: string
  description: string
  category: string
  artwork_url: string
  status: PodcastStatus
  is_explicit: boolean
  published_at: string | null
  created_at: string
  updated_at: string
}

export type PodcastEpisode = {
  id: number
  podcast_id: number
  title: string
  slug: string
  description: string
  season_number: number
  episode_number: number | null
  episode_type: PodcastEpisodeType
  audio_url: string
  artwork_url: string
  duration_ms: number
  status: PodcastEpisodeStatus
  is_explicit: boolean
  scheduled_at: string | null
  published_at: string | null
  created_at: string
  updated_at: string
}

export type PodcastDetails = {
  podcast: Podcast
  episodes: PodcastEpisode[]
}

export type CreatePodcastInput = {
  title: string
  description?: string
  category?: string
  is_explicit?: boolean
}

export type UpdatePodcastInput = {
  title?: string
  description?: string
  category?: string
  status?: PodcastStatus
  is_explicit?: boolean
}

export type CreatePodcastEpisodeInput = {
  title: string
  description?: string
  season_number?: number
  episode_number?: number
  episode_type?: PodcastEpisodeType
  duration_ms?: number
  status?: PodcastEpisodeStatus
  is_explicit?: boolean
}

export type UpdatePodcastEpisodeInput = {
  title?: string
  description?: string
  season_number?: number
  episode_number?: number
  episode_type?: PodcastEpisodeType
  duration_ms?: number
  status?: PodcastEpisodeStatus
  is_explicit?: boolean
}

export type UpdatePodcastArtworkInput = {
  artwork: File
}

export type UpdatePodcastEpisodeMediaInput = {
  audio?: File
  artwork?: File
}
